src/azure-cli/azure/cli/command_modules/acs/base_decorator.py

```diff
@@ -26,11 +26,9 @@ def validate_decorator_mode(decorator_mode) -> bool:
     :return: bool
     """
     is_valid_decorator_mode = False
-    try:
+
+    if isinstance(decorator_mode, DecoratorMode):
```

Comment:
Why do we force `decorator_mode` to be a member of `DecoratorMode`?

https://github.com/Azure/azure-cli/blob/15ea334c34e5dd8cb8ababa6b7779680042b0758/src/azure-cli/azure/cli/command_modules/acs/_consts.py#L207-L211

What will be the issue if `decorator_mode` is `1` or `2`?

Reply:
decorator_mode  is a flag, and I think it's more controllable for its value to be an enum.

if the value is set to `1`(int), it will not pass the validation.

>  https://github.com/Azure/azure-cli/blob/15ea334c34e5dd8cb8ababa6b7779680042b0758/src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_base_decorator.py#L23-L30